src/main/groovy/com/athaydes/gradle/ceylon/task/ResolveCeylonDependenciesTask.groovy

```diff
@@ -61,14 +62,29 @@ class ResolveCeylonDependenciesTask extends DefaultTask {
             }
         }
 
-        project.configurations*.resolve()
+        project.configurations.findAll { canBeResolved( (Configuration) it ) }*.resolve()
 
         def dependencyTree = dependencyTreeOf( project, moduleDeclaration )
 
         log.info( 'No dependency problems found!' )
 
         project.extensions.add( CEYLON_DEPENDENCIES, dependencyTree )
     }
+	
+	/**
+	 * Checks whether a configuration can actually be resolved. Since Gradle 3.4 this is not the case
+	 * anymore for all configurations and must be explicitely checked to avoid an exception.
+	 * 
+	 * @param configuration the configuration to check
+	 * @return {@code true} if the configuration can be resolved (this is always the case for Gradle < 3.4)
+	 */
+	static boolean canBeResolved( Configuration configuration ) {
+		MetaMethod canBeResolvedMeta = configuration.metaClass.getMetaMethod( "isCanBeResolved" );
+		if ( canBeResolvedMeta != null ) {
+			return ( boolean ) canBeResolvedMeta.invoke( configuration, new Object[0] );
```

Comment:
In Groovy, you don't need this ugly reflection stuff.

Maybe add `@CompileDynamic` to this method (to add an exception to the `@CompileStatic` at the class level) and just call the method if the method exists:

```groovy
if ( canBeResolvedMeta ) {
    return configuration.isCanBeResolved()
}
```